### PR TITLE
Feat: PKI Certs for CP/DP authentication

### PR DESCRIPTION
**Pull request by acgoldsmith**
### Description

Konnect now supports pinned PKI certs for CP/DP authentication. This means that Konnect supports digital certificates signed by a trusted CA in Konnect for CP/DP authentication.

[DOCU-3357](https://konghq.atlassian.net/browse/DOCU-3357)

### Testing instructions

Preview link: 
https://deploy-preview-6155--kongdocs.netlify.app/konnect/gateway-manager/data-plane-nodes/secure-communications/
https://deploy-preview-6155--kongdocs.netlify.app/konnect/updates/#october-2023

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

[DOCU-3357]: https://konghq.atlassian.net/browse/DOCU-3357?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by acgoldsmith** [comment]
replaces https://github.com/Kong/docs.konghq.com/pull/6152 that was created against dupe branch.

**Comment by lena-larionova** [comment]
Commenting here as well as in Slack, to make sure we're tracking info in the right place: it looks like we don't need a new entry for a PKI certs API. The functionality is going into existing entries, the Control Planes/Runtime Groups APIs: https://github.com/Kong/platform-api/pull/322/files

**Comment by acgoldsmith** [comment]
copying over comment from slack:
I also want to propose rename of the overall page. Instead of Renew Data Plane Node Certificates, maybe we call it Data Plane Node Certificates  and have a section for renew under it.

File: app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+* **Pinned mode**: This is the default mode. The same public key is added to the data plane and the control plane, and the control plane uses this public key to authenticate the data plane. Validation of the certificate only happens on the control plane.
+* **Public Key Infrastructure (PKI) mode**: This mode uses digital certificates signed by a certificate authority, or a chain of certificate authorities, to authenticate between control plane and data plane. The public key is added to the data plane, while the chain of certificate authority is added to the control plane. Any certificate authority from the chain can be used to authenticate between dataplane and control plane.  {{site.konnect_short_name}} validates both the control plane and data plane sides by checking if they are from the same certificate authority, thereby increasing the security of the network and eliminating the risks associated with transporting private keys. This mode is only supoprted for Control Plane and Control Plane Groups.
```
Comment (acgoldsmith) [inline app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md:8]:
```suggestion
* **Public Key Infrastructure (PKI) mode**: This mode uses digital certificates signed by a certificate authority, or a chain of certificate authorities, to authenticate between control plane and data plane. The public key is added to the data plane, while the chain of certificate authority is added to the control plane. Any certificate authority from the chain can be used to authenticate between dataplane and control plane.  {{site.konnect_short_name}} validates both the control plane and data plane sides by checking if they are from the same certificate authority, thereby increasing the security of the network and eliminating the risks associated with transporting private keys. This mode is only supported for Control Plane and Control Plane Groups.
```

File: app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
+* **Upload only cert2 to the control plane**: This would mean any cert coming in that has (issuer: intermediary) would be trusted. You can include just `cert1` in your data plane request. The control plane would trust any certificate issued by the intermediary public key. 
+* **Upload only cert3 to the control plane**: This is the typical PKI case. It means any cert signed by the root is trusted. However, since `cert1` is signed by an intermediary and `cert2` is signed by root, you need to include both `cert1` and `cert2` in your data plane request. The control plane would trust the whole chain because `cert2` is issued by `cert3` and `cert1` is issued by `cert2`.
+
+You can generate pinned certificates in {{site.konnect_saas}} or bring your own pinned and PKI certificates. Data plane certificates generated by {{site.konnect_saas}} expire every ten years. If you bring your own certificates, make sure to review the expiration date and associated metadata.
```
Comment (acgoldsmith) [inline app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md:24]:
```suggestion
You can generate pinned certificates in {{site.konnect_short_name}} or bring your own pinned and PKI certificates. Data plane certificates generated by {{site.konnect_short_name}} expire every ten years. If you bring your own certificates, make sure to review the expiration date and associated metadata.
```

File: app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+* **Pinned mode**: This is the default mode. The same public key is added to the data plane and the control plane, and the control plane uses this public key to authenticate the data plane. Validation of the certificate only happens on the control plane.
+* **Public Key Infrastructure (PKI) mode**: This mode uses digital certificates signed by a certificate authority, or a chain of certificate authorities, to authenticate between control plane and data plane. The public key is added to the data plane, while the chain of certificate authority is added to the control plane. Any certificate authority from the chain can be used to authenticate between dataplane and control plane.  {{site.konnect_short_name}} validates both the control plane and data plane sides by checking if they are from the same certificate authority, thereby increasing the security of the network and eliminating the risks associated with transporting private keys. This mode is only supoprted for Control Plane and Control Plane Groups.
+
+## Set certificate authentication mode on the Control Plane
```
Comment (acgoldsmith) [inline app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md:10]:
```suggestion
## Set certificate authentication mode
```

File: app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md [hunk 4]
```diff
@@ -1,15 +1,32 @@
...
+* **Upload only cert2 to the control plane**: This would mean any cert coming in that has (issuer: intermediary) would be trusted. You can include just `cert1` in your data plane request. The control plane would trust any certificate issued by the intermediary public key. 
+* **Upload only cert3 to the control plane**: This is the typical PKI case. It means any cert signed by the root is trusted. However, since `cert1` is signed by an intermediary and `cert2` is signed by root, you need to include both `cert1` and `cert2` in your data plane request. The control plane would trust the whole chain because `cert2` is issued by `cert3` and `cert1` is issued by `cert2`.
+
+You can generate pinned certificates in {{site.konnect_saas}} or bring your own pinned and PKI certificates. Data plane certificates generated by {{site.konnect_saas}} expire every ten years. If you bring your own certificates, make sure to review the expiration date and associated metadata.
+
+## Renew Data Plane Node Certificates
```
Comment (acgoldsmith) [inline app/konnect/gateway-manager/data-plane-nodes/renew-certificates.md:26]:
```suggestion
## Renew certificates
```

**Review by cloudjumpercat — APPROVED** [review]
LGTM!

**Pull request merged**